src/coreclr/jit/gentree.cpp

```diff
@@ -22380,3 +22380,78 @@ bool GenTree::IsInvariant() const
     GenTree* lclVarTree = nullptr;
     return OperIsConst() || Compiler::impIsAddressInLocal(this, &lclVarTree);
 }
+
+//------------------------------------------------------------------------
+// gtReduceStrength: reduce successive variable adds into a single multiply,
+// e.g., i + i + i + i => i * 4.
+//
+// Arguments:
+//    tree - tree for reduction
+//
+// Return Value:
+//    reduced tree if pattern matches, original tree otherwise
+//
+GenTree* Compiler::gtReduceStrength(GenTree* tree)
+{
+    // ADD(_, V0) starts the pattern match
+    if (tree->OperGet() != GT_ADD || tree->gtOverflow() || optValnumCSE_phase)
```

Comment:
```suggestion
    if (!tree->OperIs(GT_ADD) || tree->gtOverflow() || optValnumCSE_phase)
```
Nit: why not use `OperIs` everywhere?

The `optValnumCSE_phase` is redundant with the `fgGlobalMorph` guard under which the method is called. I suggest removing it and instead asserting that we are in global morph, if you don't intend to relax that restriction. If you do, then `optValnumCSE_phase` should, obviously, stay.